### PR TITLE
migrating sync calls to backend

### PR DESCRIPTION
**Pull request by NileshPant1999**

### CODE REVIEW

File: src/app/xero/dashboard/dashboard.component.html [hunk 1]
```diff
@@ -133,7 +133,14 @@ <h1 class="text-center onboarding-stepper--title">Welcome to Fyle App for Xero I
   <div class="dashboard" *ngIf="currentState === allOnboardingStates.isOnboarded && !isLoading">
     <div class="mat-h2 dashboard--header">
       Dashboard
+      <mat-icon class="dashboard--stats-card-sync"
+        mat-list-icon
+        matTooltip="Sync Fyle and NetSuite Dimensions"
```
Comment (NileshPant1999) [inline src/app/xero/dashboard/dashboard.component.html:138]:
```suggestion
        matTooltip="Sync Fyle and Xero Dimensions"
```

File: src/app/xero/dashboard/dashboard.component.ts [hunk 2]
```diff
@@ -187,24 +189,21 @@ export class DashboardComponent implements OnInit {
     });
   }
 
+  syncDimension() {
+    const that = this;
+
+    that.mappingsService.refreshFyleDimensions().subscribe(() => {});
+    that.mappingsService.refreshXeroDimensions().subscribe(() => {});
+
+    that.snackBar.open('Refreshing Fyle and Netsuite Data');
```
Comment (NileshPant1999) [inline src/app/xero/dashboard/dashboard.component.ts:198]:
```suggestion
    that.snackBar.open('Refreshing Fyle and Xero Data');
```

File: src/app/core/services/mappings.service.ts [hunk 3]
```diff
@@ -23,21 +23,60 @@ export class MappingsService {
...
+        refCount()
+      );
+    }
+    return this.destinationWorkspace;
+  }
+
+  refreshXeroDimensions() {
+    const workspaceId = this.workspaceService.getWorkspaceId();
+
+    return this.apiService.post(`/workspaces/${workspaceId}/xero/refresh/`, {});
```
Comment (ashwin1111) [inline src/app/core/services/mappings.service.ts:69]: can we name it `refresh_dimensions`, please change it everywhere

File: src/app/xero/dashboard/dashboard.component.scss [hunk 4]
```diff
@@ -112,6 +112,15 @@ $stats-desc: #545454;
         color: $fy-dashboard-stat-icon;
       }
 
+      &-sync {
+          color: $fy-dashboard-stat-icon;
```
Comment (ashwin1111) [inline src/app/xero/dashboard/dashboard.component.scss:116]: indent

**Review by Sravanksk — APPROVED** [review]

**Pull request merged**